### PR TITLE
[blog]: How to add blog post to OSCSA blog

### LINKED ISSUE #44: [7/12/2022] - Documentation on how to make blog posts
### Description
Add documentation on how to add blog posts to the [OSCSA Blog](https://oscsa-en-blog.netlify.app/)

This is meant to help newcomers to create a post in the OSCSA https://oscsa-en-blog.netlify.app/. It assumes that they have written a draft and would like to add it to the website. Please make it as easy as possible to newcomers who have limited knowledge in Open source or Quarto or even programming.

### PR DESCRIPTION
**Pull request by RealRichi3**
### Description
Add documentation on Quarto setup, installation guide, and creating blogposts.

Fixes #44

### Changes proposed
- [x] *Add documentation on how to setup quarto locally*
- [x] *Add installation guide*
- [x] *Add steps to write new blog posts using quarto*

### CODE REVIEW

**Comment by RealRichi3** [comment]
@sandygudie this is a draft request for the documentation on how to create a blog post, I have added the installation guide and relevant links.

It's not complete yet

**Comment by BatoolMM** [comment]
This looks wonderful, thank you so much 😊. I will add a few comments after our meeting today!

**Comment by RealRichi3** [comment]
Thanks @BatoolMM

**Comment by RealRichi3** [comment]
I'd like to add better images

File: posts/How-to-add-blog-post/index.qmd [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+Quarto is a new open source project that aims to make it easier for people to create and publish blog posts. It is a static site generator that uses the Markdown syntax to create blog posts. Quarto is written in the R programming language and is available for download on GitHub.
+
+## Get started
+### Prerequisites
+To get started with contributing with Quarto, you will need;
+
+- A GitHub account.
+- A basic understanding of Markdown is recommended but not necessary.
+
+### Install Quarto
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:39]: I don't think they need to install Quarto - so maybe this paragraph can be moved to the side using `{.aside}`

File: posts/How-to-add-blog-post/index.qmd [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+it should follow the format `[blog]: Title of your blog`
+
+Automatically, the PR template will be added to the description of your pull request. You can add more information about your pull request in the description.
+
+#
+
+##### Step 8 - Merge your changes
+Once you have created a pull request, you can wait for your changes to be reviewed and merged to the main branch.
+
+#
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:190]: Can you show here a sample of a blog post with YAML (very small sample, what the end product looks like?)

File: posts/How-to-add-blog-post/index.qmd [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+### Preview your blog post
+Since we are working directly from the Github account and Quarto is a command line tool, you can not preview the HTML rendered version of your blogpost.
+To see the preview:
+
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:195]: Just show how they can preview it from GitHub without the command line!

**Review by BatoolMM — CHANGES_REQUESTED** [review]
This si wonderful, thank you lots, only few suggestions!

File: posts/How-to-add-blog-post/index.qmd (hunk 1 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:39]: Oh, alright thanks

File: posts/How-to-add-blog-post/index.qmd (hunk 3 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:195]: I didn't know it is possible to preview without using the terminal

Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:195]: I'll check the documentation on how to do it

File: posts/How-to-add-blog-post/index.qmd (hunk 2 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:190]: Alright

File: posts/How-to-add-blog-post/index.qmd (hunk 1 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:40]: done

File: posts/How-to-add-blog-post/index.qmd (hunk 2 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:191]: by sample do you mean the rendered preview?

File: posts/How-to-add-blog-post/index.qmd (hunk 3 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:233]: I used the netlify preview link instead, is this okay?

File: posts/How-to-add-blog-post/index.qmd (hunk 2 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:205]: I think i get it now, you meant a YAML code preview right

File: posts/How-to-add-blog-post/index.qmd [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+#
+
+## What is Quarto?
+Quarto is a new open source project that aims to make it easier for people to create and publish blog posts. It is a static site generator that uses the Markdown syntax to create blog posts. Quarto is written in the R programming language and is available for download on GitHub.
+
+## Get started
+### Prerequisites
+To get started with contributing with Quarto, you will need;
+
+- A GitHub account.
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:36]: Add a link to the GitHub
```suggestion
- A [GitHub](https://github.com/) account.
```

File: posts/How-to-add-blog-post/index.qmd [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+- A basic understanding of Markdown is recommended but not necessary.
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:37]: The same with Markdown, need to link it to any cheatsheet for Markdown.
```suggestion
- A basic understanding of Markdown is recommended but not necessary.
```

File: posts/How-to-add-blog-post/index.qmd [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+:::
+#
+
+### Setup Local environment
+To start making contributions to the OSCA blog, you will need to setup a personal environment on your personal Github account. This will allow you to make changes to the blog without affecting the live version of the blog.
+
+#
+
+##### Step 1 - Fork the repository
+To get started, you will need to fork the OSCA blog repository from GitHub.
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:53]: Add links
```suggestion
To get started, you will need to fork the [OSCA blog repository](https://github.com/Open-Science-Community-Saudi-Arabia/OSCSA-en-blog/) from GitHub.
```

File: posts/How-to-add-blog-post/index.qmd [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+categories: 
+    - "blog"
+    - "quarto"
+    - "Outreachy"
+description: "This is a tutorial on how to add a blog post to the OSCA blog using Quarto."
+licence: "CC BY"
+type: "post"
+toc-title: "Table of Contents"
+toc-location: right
+toc-depth: 3
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:15]:
```suggestion
toc-depth: 5
```

File: posts/How-to-add-blog-post/index.qmd [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+```
+
+To see a preview of your blog post, you can visit the netlify preview link that was generated when you created the pull request. 
+
+![](./preview_link.png){fig-align="center" width="80%"}
+
+#
+
+### Conclusion
+In this tutorial, we learned how to create a blog post using Quarto. We started by installing Quarto and setting up our local environment. We then created a new blog post using the Quarto syntax and previewed it using the Quarto serve command. Finally, we submitted a pull request for the new changes.
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:241]:
```suggestion
In this tutorial, we learned how to create a blog post using Quarto. We started by installing Quarto and setting up our local environment. We then created a new blog post using the Quarto syntax. Finally, we submitted a pull request for the new changes and previewed it using the Netlify Preview.
```

**Review by BatoolMM — CHANGES_REQUESTED** [review]

**Review by sandygudie — COMMENTED** [review]
Welldone @RealRichi3 
I will like to work on the images, I think using arrows and squares rather than free-hand would look great example
image

**Comment by sandygudie** [comment]
@RealRichi3 @BatoolMM  images updated
Please check review

File: posts/How-to-add-blog-post/index.qmd (hunk 3 above)
Comment (BatoolMM) [reply posts/How-to-add-blog-post/index.qmd:236]: Yes, that's what I meant, thank you!

**Comment by BatoolMM** [comment]
Thank you both, it looks amazing, but some of the images isn't showing in the preview!

**Comment by sandygudie** [comment]
> Thank you both, it looks amazing, but some of the images isn't showing in the preview!

Ohkay, I see it too, I will look into it

**Comment by RealRichi3** [comment]
Looks good @sandygudie, but some images aren't displaying

**Review by sandygudie — COMMENTED** [review]
The images works now
@BatoolMM  @RealRichi3 please review

@RealRichi3 there are too many images in the folder, please filter out the ones you don't need

**Comment by RealRichi3** [comment]
> The images works now @BatoolMM @RealRichi3 please review
> 
> @RealRichi3 there are too many images in the folder, please filter out the ones you don't need

done.

File: posts/How-to-add-blog-post/index.qmd (hunk 5 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:37]: done

File: posts/How-to-add-blog-post/index.qmd (hunk 4 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd]: done

File: posts/How-to-add-blog-post/index.qmd (hunk 6 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd]: @sandygudie had fixed this

**Comment by RealRichi3** [comment]
@sandygudie @BatoolMM, does everything look okay now?

**Comment by sandygudie** [comment]
> The images works now @BatoolMM @RealRichi3 please review
> 
> @RealRichi3 there are too many images in the folder, please filter out the ones you don't need

Welldone @RealRichi3

**Review by sandygudie — APPROVED** [review]
Welldone Richie
It looks great for me

File: posts/How-to-add-blog-post/index.qmd [hunk 9]
```diff
@@ -0,0 +1,253 @@
+---
+title: "How to add a blog post using Quarto"
```
Comment (BatoolMM) [inline posts/How-to-add-blog-post/index.qmd:2]:
```suggestion
title: "How to create a blog post in the Open Science Community Saudi Arabia using Quarto?"
```

**Review by BatoolMM — CHANGES_REQUESTED** [review]

**Comment by BatoolMM** [comment]
@alswajiab would you mind to review it, please?

**Review by alswajiab — CHANGES_REQUESTED** [review]
Thanks, @RealRichi3, I suggest adding a table of contents to the post to make it more readable.

File: posts/How-to-add-blog-post/index.qmd (hunk 9 above)
Comment (RealRichi3) [reply posts/How-to-add-blog-post/index.qmd:2]: alright, noted, i'll make the necessary changes

**Comment by RealRichi3** [comment]
> Thanks, @RealRichi3, I suggest adding a table of contents to the post to make it more readable.

Alright @alswajiab

**Comment by RealRichi3** [comment]
> Thanks, @RealRichi3, I suggest adding a table of contents to the post to make it more readable.

@alswajiab there's already something like a table of content on the right side of the blog post, is that okay, or should I do something bigger?

**Comment by BatoolMM** [comment]
@RealRichi3 The CI isn't passing!

**Comment by BatoolMM** [comment]
@all-contributors please add @RealRichi3 for blog posts

**Comment by BatoolMM** [comment]
@all-contributors please add @alswajiab for review

**Comment by RealRichi3** [comment]
> @RealRichi3 The CI isn't passing!

Ohh, i think you just fixed it

**Review by BatoolMM — CHANGES_REQUESTED** [review]
Few comments:
- I fixed the CI, make sure the CI is passing.
- There are multiple images added but not used, can you remove them.
- Add alternative tags to the images to make them accessible for screenreader.
- Add one of the [following image](https://allisonhorst.com/cetinkaya-rundel-lowndes-quarto-keynote) as thumbnail

**Comment by RealRichi3** [comment]
> Few comments:
> 
> * I fixed the CI, make sure the CI is passing.
> * There are multiple images added but not used, can you remove them.
> * Add alternative tags to the images to make them accessible for screenreader.
> * Add one of the [following image](https://allisonhorst.com/cetinkaya-rundel-lowndes-quarto-keynote) as thumbnail

@BatoolMM for the CI please how did you fix it, the issue comes up from time to time

**Comment by RealRichi3** [comment]
For the thumbnail, should i use just one of the images, or all of them

**Comment by BatoolMM** [comment]
> For the thumbnail, should i use just one of the images, or all of them

Just one, feel free to choose any one!

**Comment by BatoolMM** [comment]
> > Few comments:
> > 
> > * I fixed the CI, make sure the CI is passing.
> > * There are multiple images added but not used, can you remove them.
> > * Add alternative tags to the images to make them accessible for screenreader.
> > * Add one of the [following image](https://allisonhorst.com/cetinkaya-rundel-lowndes-quarto-keynote) as thumbnail
> 
> @BatoolMM for the CI please how did you fix it, the issue comes up from time to time

You have to look at the log file of the build o find out what the error is. It's usually something to do with YAML not lining.

**Comment by RealRichi3** [comment]
> usually

Oh, alright then

**Comment by RealRichi3** [comment]
image

@BatoolMM i tried to add the alt image tags, but they text keeps showing, is there a way to hide it?, I checked online I think the only way is to use CSS to hide the text, but this defeats the whole purpose of adding the alt texts

**Comment by BatoolMM** [comment]
@RealRichi3 The ey to add alt tags in Markdown is using this syntax `alt`. You can read more [here](https://quarto.org/docs/authoring/figures.html). Let me know if it's still not clear. I also added the Quarto VS Code extension into gitpod of this repo in case it can help you!

**Comment by RealRichi3** [comment]
@BatoolMM thanks, i just saw the correct way to do it, `{fig-alt="A drawing of an elephant."}`

**Comment by RealRichi3** [comment]
@BatoolMM do you prefer if the alt text if visible under each image? or should I leave it just the way it is?

**Comment by BatoolMM** [comment]
> https://deploy-preview-46--oscsa-en-blog.netlify.app

Thank you for pointing this out, Markdown use this syntax `alt` but Quarto uses `caption{alt}`. Alternative tags should always be visible and only accessed by the screen reader.

**Comment by RealRichi3** [comment]
> > https://deploy-preview-46--oscsa-en-blog.netlify.app
> 
> Thank you for pointing this out, Markdown use this syntax `alt` but Quarto uses `caption{alt}`. Alternative tags should always be **invisible** and only accessed by the screen reader.

Alright, thanks @BatoolMM, I'll work on the thumbnails next

**Comment by RealRichi3** [comment]
@BatoolMM, i tried to add the thumbnail, but I didn't really see any clear guide, here's what I tried
1. I tried to embed meta tags for the thumbnail, just like it's done in HTML documents, so I set the `embed-resources: true`
I thought this will work but it keeps showing the code, instead of adding the thumbnail

This is the YAML and the HTML code I tried
---
title: "All you need to create a blog post in the Open Science Community Saudi Arabia using Quarto?"
author: 
    name: "Richie Moluno, Goodnews Sandy"
date: "2022-12-08"
categories: 
    - "blog"
    - "quarto"
    - "outreachy"
description: "This is a step-by-step guide on how to create your first a blog post without any coding."
licence: "CC BY"
type: "post"
format: 
  html:
    embed-resources: true
    df-print: paged
    toc: true
    toc-depth: 5
    toc-location: left
    toc-title: "On this page"
image: "./quarto-logo.jpg"
thumbnail: "./thumbnail.png"
citation: 
    url: https://oscsa.netlify.app/posts/how-to-add-a-blog-post/
site-url: https://oscsa.netlify.app/posts/how-to-add-a-blog-post/
---


{fig-align="center" fig-caption="Quarto logo"}

```html
<meta property=’og:image’ content='./thumbnail.png'/>
<meta property='og:image:width' content='1200' />
<meta property='og:image:height' content='627' />
<meta property="og:type" content='article'/>
```
image

2. For the second approach, in the YAML I just set thumbnail to the image path, none worked

**Comment by RealRichi3** [comment]
> @RealRichi3 did you just try to set `image: "thumbnail.jpg"`? `

I tried `thumbnail: ./thumbnail.png`

**Comment by BatoolMM** [comment]
Thank you @RealRichi3 for changing the thumbnail, I was looking at the preview - and there're a few things changed in the main blog website. Would you be able to have a look at it?

**Comment by RealRichi3** [comment]
Alright @BatoolMM I'll check it out

**Comment by sandygudie** [comment]
Hi @RealRichi3  the image is missing ,
image

**Comment by sandygudie** [comment]
I noticed the deployed preview for the home site was off so I made a push(fix the missing image) from my end to see if that would fix it. it works now

**Comment by RealRichi3** [comment]
> I noticed the deployed preview for the home site was off so I made a push(fix the missing image) from my end to see if that would fix it. it works now

Thanks @sandygudie

**Review by BatoolMM — APPROVED** [review]
Thank you @RealRichi3 and @sandygudie. It looks wonderful!

**Comment by BatoolMM** [comment]
@all-contributors please add @sandygudie for blog, docs

**Comment by BatoolMM** [comment]
@all-contributors please add @RealRichi3 for blog, docs

**Comment by RealRichi3** [comment]
Thanks @BatoolMM

**Pull request merged**